include/git2/errors.h

```diff
@@ -55,6 +55,7 @@ typedef enum {
 	GIT_ITEROVER        = -31,	/**< Signals end of iteration with iterator */
 	GIT_RETRY           = -32,	/**< Internal only */
 	GIT_EMISMATCH       = -33,	/**< Hashsum mismatch in object */
+	GIT_EINDEXDIRTY     = -34,  /**< Unsaved changes in the index would be overwritten */
```

Comment:
This time I'm fast to review, so I'm allowed to nag at the whitespace error here :P